Hestia/app/src/main/java/com/rugged/application/hestia/Activator.java

```diff
@@ -0,0 +1,49 @@
+package com.rugged.application.hestia;
+
+public class Activator<T> {
+    private int activatorId;
+    private T state;
+    private String name;
+    private String stateType;
```

Comment:
We still need to find a proper OO solution to this problem. This is hard because we receive a JSON file containing a generic type for the state. We cannot infer this type, except if we use a switch statement based on the stateType. I will create an issue for this.